packages/block-editor/src/store/selectors.js

```diff
@@ -2294,8 +2294,12 @@ const checkAllowListRecursive = ( blocks, allowedBlockTypes ) => {
 function getUserPatterns( state ) {
 	const userPatterns =
 		state?.settings?.__experimentalReusableBlocks ?? EMPTY_ARRAY;
-	const { patternCategoriesMap: categories } =
-		state?.settings?.__experimentalUserPatternCategories ?? {};
+	const userPatternCategories =
+		state?.settings?.__experimentalUserPatternCategories ?? [];
+	const categories = new Map();
+	userPatternCategories?.forEach( ( userCategory ) =>
```

Comment:
Nit: needless optional chaining:

```suggestion
	userPatternCategories?forEach( ( userCategory ) =>
```

Reply:
thanks, fixed.